tools/src/main/java/org/apache/kafka/tools/ConsoleShareConsumerOptions.java

```diff
@@ -0,0 +1,486 @@
+                .ofType(String.class);
+        maxMessagesOpt = parser.accepts("max-messages", "The maximum number of messages to consume before exiting. If not set, consumption is continual.")
+                .withRequiredArg()
+                .describedAs("num_messages")
+                .ofType(Integer.class);
+        timeoutMsOpt = parser.accepts("timeout-ms", "If specified, exit if no message is available for consumption for the specified interval.")
+                .withRequiredArg()
+                .describedAs("timeout_ms")
+                .ofType(Integer.class);
+        rejectMessageOnErrorOpt = parser.accepts("reject-message-on-error", "If there is an error when processing a message, " +
+                "reject it instead of halt.");
+        bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED: The server(s) to connect to.")
+                .withRequiredArg()
+                .describedAs("server to connect to")
+                .ofType(String.class);
+        keyDeserializerOpt = parser.accepts("key-deserializer")
+                .withRequiredArg()
+                .describedAs("deserializer for key")
+                .ofType(String.class);
+        valueDeserializerOpt = parser.accepts("value-deserializer")
+                .withRequiredArg()
+                .describedAs("deserializer for values")
+                .ofType(String.class);
+        groupIdOpt = parser.accepts("group", "The consumer group id of the consumer.")
+                .withRequiredArg()
+                .describedAs("consumer group id")
+                .ofType(String.class);
+        // TODO : Add --release and --reject option
+
+        try {
+            options = parser.parse(args);
+        } catch (OptionException oe) {
+            CommandLineUtils.printUsageAndExit(parser, oe.getMessage());
+        }
+
+        CommandLineUtils.maybePrintHelpOrVersion(this, "This tool helps to read data from Kafka topics using consumer share groups and outputs it to standard output.");
+
+        checkRequiredArgs();
+
+        Properties consumerPropsFromFile = options.has(consumerConfigOpt)
+                ? Utils.loadProps(options.valueOf(consumerConfigOpt))
+                : new Properties();
+        Properties extraConsumerProps = CommandLineUtils.parseKeyValueArgs(options.valuesOf(consumerPropertyOpt));
+
+        Set<String> groupIdsProvided = checkConsumerGroup(consumerPropsFromFile, extraConsumerProps);
+        consumerProps = buildConsumerProps(consumerPropsFromFile, extraConsumerProps, groupIdsProvided);
+        formatter = buildFormatter();
+    }
+
+    private void checkRequiredArgs() {
+        if (!options.has(topicOpt)) {
+            CommandLineUtils.printUsageAndExit(parser, "--topic is a required argument");
+        }
+        CommandLineUtils.checkRequiredArgs(parser, options, bootstrapServerOpt);
+    }
+
+    private Set<String> checkConsumerGroup(Properties consumerPropsFromFile, Properties extraConsumerProps) {
```

Comment:
Because the command-line argument has a default of "share", we know that there will be at least one provided. I suppose the check that they're all the same still makes sense. I think the way to achieve this is actually to say that if there are no group IDs provided, then apply the default of "share". Otherwise, the overriding by the properties and file is troublesome.

Also, this method should be `checkShareGroup`.